engine/src/test/java/org/camunda/bpm/engine/test/history/HistoricTaskInstanceTest.java

```diff
@@ -1132,4 +1132,28 @@ public void testQueryByProcessInstanceBusinessKeyAndArray() {
     assertEquals(0, query.processInstanceBusinessKeyIn(businessKey1, businessKey2).processInstanceBusinessKey(unexistingBusinessKey).count());
     assertEquals(1, query.processInstanceBusinessKeyIn(businessKey2, businessKey3).processInstanceBusinessKey(businessKey2).count());
   }
+
+  @Deployment(resources = { "org/camunda/bpm/engine/test/api/history/multiprocess/rootProcess.bpmn",
+      "org/camunda/bpm/engine/test/api/history/multiprocess/secondLevelProcess.bpmn",
+      "org/camunda/bpm/engine/test/api/history/multiprocess/thirdLevelProcess.bpmn" })
+  @Test
+  public void testQueryByRootProcessInstanceId() {
+    // given process definitions
+
+    // when
+    String rootProcessId = runtimeService.startProcessInstanceByKey("root-process").getId();
+    runtimeService.startProcessInstanceByKey("process-3");
+    while (historyService.createHistoricProcessInstanceQuery().unfinished().count() > 0) {
+      taskService.createTaskQuery().list().forEach(task -> taskService.complete(task.getId()));
```

Comment:
🙃 You could've used processes without user tasks, so they complete without any extra steps.
Your solution works too, but needs a bit more bpmn & code.

Reply:
I originally tried with script tasks because of this, but they weren't being returned by the API. Unless I'm mistaken, the API only returns user tasks.

Reply:
You're right, it's a `HistoricTaskInstanceQuery`, my bad!